Automattic/jetpack
# Payment Button Block: Use the shared ProductManagementControls

Fixes #23414
Fixes #23506

#### Changes proposed in this Pull Request:

* We now use the [Product Management Controls component](https://github.com/Automattic/jetpack/issues/23412) instead of having ad-hoc implementations of product management for each payment block.
* Cleaned up the code and refactored the block to a functional component.
* Removed the initial form for consistency between our different payment blocks.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create or edit a Post.
* Add a Payments Block to the Post.
* You should not see a form above the button indicating to choose a plan.
* You should see the Product Management Controls.
* Change the subscription and save.
* Reload the editor, the subscription should be the one you've manually set.

You should also test a site that is on a free plan. In this case, you should see the upgrade nudge.

Also, test the Premium Content block. In this scenario, we should not see the membership controls for the main button, but any other Payment Button added should continue to work as expected.


#### Testing evidence, happy case
https://user-images.githubusercontent.com/1989914/159494023-9c5d7f82-6023-46f6-9356-a2bc4304746c.mp4


#### Testing evidence, Premium Content block nesting

https://user-images.githubusercontent.com/1989914/159494507-32b5ffb3-103b-4ecd-9d56-ec39a4fdc521.mp4

#### Testing evidence, Payments button on a free plan
https://user-images.githubusercontent.com/1989914/159495852-a8118fc0-6c31-4f4e-bdda-e0ccd426bbe3.mp4
